src/test/java/org/jabref/gui/autosaveandbackup/BackupManagerGitTest.java

```diff
@@ -0,0 +1,166 @@
+
+package org.jabref.gui.autosaveandbackup;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.jabref.gui.LibraryTab;
+import org.jabref.logic.FilePreferences;
+import org.jabref.logic.preferences.CliPreferences;
+import org.jabref.model.database.BibDatabase;
+import org.jabref.model.database.BibDatabaseContext;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.BibEntryTypesManager;
+import org.jabref.model.metadata.MetaData;
+
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class BackupManagerGitTest {
+
+    private Path tempDir1;
+    private Path tempDir2;
+    private Path tempDir;
+    private LibraryTab mockLibraryTab;
+    private BibDatabaseContext mockDatabaseContext1;
+    private BibDatabaseContext mockDatabaseContext2;
+    private BibEntryTypesManager mockEntryTypesManager;
+    private CliPreferences mockPreferences;
+    private Path dataBasePath1;
+    private Path dataBasePath2;
+
+    @BeforeEach
+    public void setUp(@TempDir Path tempDir) throws IOException, GitAPIException {
+
+        // creating Entries
+        List<BibEntry> entries1 = new ArrayList<>();
+        entries1.add(new BibEntry("this"));
+        entries1.add(new BibEntry("is"));
```

Comment:
Add fields to the entries using `.withField(...)`
Example - https://github.com/JabRef/jabref/blob/38576bb828c3807bbeb7c0f1e4dde3aa5f92a08c/src/test/java/org/jabref/logic/openoffice/oocsltext/CSLFormatUtilsTest.java#L365-L375

Reply:
It is modified accordingly !